src/compiler/driver.cc

```diff
@@ -343,11 +328,43 @@ hilti::Result<hilti::Nothing> Driver::compile() {
     return hilti::Nothing();
 }
 
-hilti::Result<UnitInfo> Driver::lookupUnit(const hilti::ID& unit) {
-    if ( auto x = _units.find(unit); x != _units.end() )
+hilti::Result<TypeInfo> Driver::lookupType(const hilti::ID& id) {
+    if ( auto x = _types.find(id); x != _types.end() )
         return x->second;
     else
-        return hilti::result::Error("unknown unit");
+        return hilti::result::Error(hilti::util::fmt("unknown type '%s'", id));
+}
+
+std::vector<TypeInfo> Driver::types() const {
+    std::vector<TypeInfo> result;
+    result.reserve(_types.size());
+
+    for ( const auto& t : _types )
+        result.push_back(t.second);
+
+    return result;
+}
+
+std::vector<std::pair<TypeInfo, hilti::ID>> Driver::exportedTypes() const {
+    std::vector<std::pair<TypeInfo, hilti::ID>> result;
+
+    for ( const auto& export_ : _glue->exportedIDs() ) {
+        auto spicy_id = std::get<0>(export_);
+        auto zeek_id = std::get<1>(export_);
+
```

Comment:
We don't need copies for `spicy_id` and `zeek_id` here, let's either use `const auto&` here or a structured binding in the loop header:

```suggestion
    for ( const auto& [spicy_id, zeek_id, _] : _glue->exportedIDs() ) {
```